scwidgets/widgets.py

```diff
@@ -0,0 +1,230 @@
+        """
+        try:
+            kwargs = self._retrive_kwargs()
+        except ValueError as e:
+            print(f'Invalid inputs: {e}')
+            return
+
+        return self.callable(**kwargs)
+
+
+def _repr_html_(scope):
+    """
+    Helper method to display the scipp objects in a given scope.
+    """
+    from IPython import get_ipython
+    ipython = get_ipython()
+    out = ''
+    for category in ['Variable', 'DataArray', 'Dataset']:
+        names = ipython.magic(f"who_ls {category}")
+        out += f"<details open=\"open\"><summary>{category}s:"\
+               f"({len(names)})</summary>"
+        for name in names:
+            html = sc.table_html.make_html(eval(name, scope))
+            out += f"<details style=\"padding-left:2em\"><summary>"\
+                   f"{name}</summary>{html}</details>"
+        out += "</details>"
+    display(HTML(out))
+
+
+class PlotWidget(widgets.Box):
+    """
+    Wraps scipp.plot.plot attempting to display the result of evaluating
+    the input string as a scipp plot.
+    """
+    def __init__(self, scope: MutableMapping[str, Any]):
+        """
+        Parameters:
+        scope (dict): the scope dict to use when evaluating user input.
+        """
+        super().__init__()
+        self.scope = scope
+
+        self._data_selector = widgets.Combobox(placeholder='Data to plot')
+
+        self._button = widgets.Button(description='Plot')
+        self._button.on_click(self._on_plot_button_clicked)
+
+        self.plot_options = widgets.Output()
+
+        self.update_button = widgets.Button(description='Update')
+        self.update_button.on_click(self._on_update_button_clicked)
+
+        self.output = widgets.Output(width='100%', height='100%')
+
+        self.children = [
+            widgets.VBox([
+                widgets.HBox([self.plot_options, self.update_button]),
+                widgets.HBox([self._data_selector, self._button]), self.output
+            ])
+        ]
+
+        self.update()
+
+    def _on_plot_button_clicked(self, button):
+        self.output.clear_output()
+        with self.output:
+            display(plot(eval(self._data_selector.value, self.scope)))
+
+    def _on_update_button_clicked(self, button):
+        self.update()
+
+    def update(self):
+        """
+        Updates the display of available scipp objects
+        """
+        options = [
+            key for key, item in self.scope.items()
+            if isinstance(item, scipp_object)
+        ]
+        self._data_selector.options = options
+        self.plot_options.clear_output()
+        with self.plot_options:
+            _repr_html_(self.scope)
+
+
+# Method to hide code blocks taken from
+# https://stackoverflow.com/questions/27934885/how-to-hide-code-from-cells-in-ipython-notebook-visualized-with-nbviewer
+javascript_functions = {False: "hide()", True: "show()"}
+button_descriptions = {False: "Show code", True: "Hide code"}
+
+
+def toggle_code(state):
```

Comment:
This hides code of all cells, is there a way to just hide the code from a single cell, e.g., only those used to spawn a widget?

Reply:
There should be I'll look into the nicest way todo this.